demo/src/MemoryModelsUserInput.tsx

```diff
@@ -66,29 +73,55 @@ function MemoryModelsFileInput(props: MemoryModelsFileInputPropTypes) {
 
     const onLoadButtonClick = () => {
         props.setTextData(uploadedFileString);
+        setOpen(false);
     };
 
     return (
-        <Stack direction={"row"} spacing={2}>
-            <Input
-                type="file"
-                onChange={onChange}
-                inputProps={{
-                    accept: "application/JSON",
-                    "data-testid": "file-input",
-                }}
-                disableUnderline={true}
-            />
-            <Button
-                data-testid="file-input-reapply-button"
-                variant="contained"
-                disabled={!uploadedFileString}
-                onClick={onLoadButtonClick}
-                sx={{ textTransform: "none" }}
-            >
-                Load file data
+        <div>
+            <Button onClick={handleOpen} sx={{ textTransform: "none" }}>
+                File Input
             </Button>
-        </Stack>
+            <Modal
+                open={open}
+                onClose={handleClose}
+                data-testid="file-input-modal"
+            >
+                <Paper
+                    sx={{
```

Comment:
I'm not sure why any of these custom styles are necessary, doesn't the `Modal` component take care of the layout for us?

Reply:
The default styling for the modal was pretty much unusable:

![image](https://github.com/user-attachments/assets/714bda7d-52c2-499b-9271-be163a95de2e)

As far as I can tell, the MUI `Modal` component doesn't actually have useful style presets. The examples in their documentation all set the CSS manually. (I did find something called Joy UI, which seems to be an offshoot of MUI that does offer good default Modal styling, but I think that's a separate library we'd have to install.)

Reply:
@yoonieaj aha, thanks for clarifying. This prompted me to do some more digging in the MUI documentation, since this seemed counter-intuitive to me. If you look at the documentation for [`Modal`](https://mui.com/material-ui/react-modal/), it includes the following warning:

> The term "modal" is sometimes used to mean "dialog", but this is a misnomer. A modal window describes parts of a UI. An element is considered modal if [it blocks interaction with the rest of the application](https://en.wikipedia.org/wiki/Modal_window).

and

> If you are creating a modal dialog, you probably want to use the [Dialog](https://mui.com/material-ui/react-dialog/) component rather than directly using Modal. Modal is a lower-level construct that is leveraged by the following components...

So, I think we should be using `Dialog` here instead of `Modal`!